### PR TITLE
Fix revocation url #2

### PR DESCRIPTION
**Pull request by p53**
# Title

## Type

[] Bug fix
[] Feature request
[] Enhancement
[] Docs

## Checklist:

- [ ] My change requires a change to the documentation or CHANGELOG.
- [ ] I have updated the documentation/CHANGELOG accordingly.

### CODE REVIEW

File: handlers.go [hunk 1]
```diff
@@ -403,7 +404,7 @@ func (r *oauthProxy) logoutHandler(w http.ResponseWriter, req *http.Request) {
 	}
 
 	// set the default revocation url
-	revokeDefault := fmt.Sprintf("%s/protocol/openid-connect/logout", strings.TrimSuffix(r.config.DiscoveryURL, "/.well-known/openid-configuration"))
+	revokeDefault := fmt.Sprintf("%s/protocol/openid-connect/revoke", strings.TrimSuffix(r.config.DiscoveryURL, "/.well-known/openid-configuration"))
 	revocationURL := defaultTo(r.config.RevocationEndpoint, revokeDefault)
```
Comment (ackerleytng) [inline handlers.go:408]: Can we move these down to just above line 429 where we use `revocationURL`?

File: handlers.go [hunk 2]
```diff
@@ -438,7 +439,14 @@ func (r *oauthProxy) logoutHandler(w http.ResponseWriter, req *http.Request) {
 		encodedSecret := url.QueryEscape(r.config.ClientSecret)
 
 		// step: construct the url for revocation
-		request, err := http.NewRequest(http.MethodPost, revocationURL, bytes.NewBufferString(fmt.Sprintf("refresh_token=%s", identityToken)))
+		request, err := http.NewRequest(
+			http.MethodPost,
+			revocationURL,
+			bytes.NewBufferString(
+				fmt.Sprintf("token=%s", identityToken),
```
Comment (ackerleytng) [inline handlers.go:446]: Could we also provide `token_type_hint=refresh_token` as described here https://tools.ietf.org/html/rfc7009#section-2.1?

**Review by ackerleytng — COMMENTED** [review]
Was this tested against Keycloak? I do see that it is compliant with RFC 7009 though.

File: handlers.go (hunk 2 above)
Comment (p53) [reply handlers.go:446]: identityToken can be refresh/access token, but keycloak only supports refresh/offline tokens, IDP according RFC should infer token type also without hint so it is not necessary

**Comment by p53** [comment]
> Was this tested against Keycloak? I do see that it is compliant with RFC 7009 though.

i tested it, it works with refresh token

File: handlers.go (hunk 1 above)
Comment (p53) [reply handlers.go]: ok

**Review by ackerleytng — APPROVED** [review]
Thanks for clarifying!

**Comment by p53** [comment]
@thomas-jakemeyn @patrickleet @ryanc-me should be fixed in 1.2.1-rc1

**Pull request merged**